python/lib/core/dmod/core/data_domain_detectors.py

```diff
@@ -0,0 +1,468 @@
+        """
+
+    def is_registered(self, entry: Union[str, Type[ItemDataDomainDetector]]) -> bool:
+        """
+        Whether this is a registered subclass or the registered name of one.
+
+        Parameters
+        ----------
+        entry: Union[str, Type[ItemDataDomainDetector]]
+            The potential registration name or subclass type.
+
+        Returns
+        -------
+        bool
+            Whether this is a registered subclass or the registered name of one.
+        """
+        return (entry if isinstance(entry, str) else entry.get_registration_name()) in self._detectors
+
+    def get_all_names(self) -> List[str]:
+        """
+        Get registration names for all registered subclasses.
+
+        Returns
+        -------
+        List[str]
+            Registration names for all registered subclasses.
+        """
+        return list(self._detectors.keys())
+
+    def get_for_format(self, data_format: DataFormat) -> List[Type[ItemDataDomainDetector]]:
+        """
+        Get a list (sorted by registration name) of the detector subclasses associated with the given format.
+
+        Parameters
+        ----------
+        data_format: DataFormat
+            The data format of interest.
+
+        Returns
+        -------
+        List[Type[ItemDataDomainDetector]]
+            The sorted detector subclasses associated with the given format.
+        """
+        return [self._detectors[name] for name in sorted(self._formats_to_detectors[data_format])]
+
+    def get_for_name(self, name: str) -> Type[ItemDataDomainDetector]:
+        """
+        Get the registered subclass for the given registration name.
+
+        Parameters
+        ----------
+        name: str
+            The subclass registration name.
+
+        Returns
+        -------
+        Type[ItemDataDomainDetector]
+            The registered subclass for the given registration name.
+
+        Raises
+        ------
+        KeyError
+            Raised if this is not a valid, recognized registration name.
+        """
+        return self._detectors[name]
+
+    def register(self, subclass: Type[ItemDataDomainDetector]):
+        """
+        Register the given subclass of :class:`ItemDataDomainDetector`.
+
+        Parameters
+        ----------
+        subclass: Type[ItemDataDomainDetector]
+            A subclass of :class:`ItemDataDomainDetector`.
+
+        Notes
+        -----
+        If an already-registered subclass is passed in another call to this method, nothing will happen.  The instance's
+        state will not change, nor will an error be thrown, and the method will quietly return.
+
+        Raises
+        ------
+        DmodRuntimeError
+            If the registration name is already in use for a different subclass.
+        """
+        name = subclass.get_registration_name()
+        if name not in self._detectors:
+            self._detectors[name] = subclass
+            if subclass.get_data_format() is not None:
```

Comment:
Hmm, probably not. I considered linking the type to the generic format. I didn't, but I am rethinking it. 

It would be useful for anything that can get the registry to be able to get at the universal detector. But maybe that only really works if it can ask the registry for a detector without knowing the format.

Reply:
Yeah, the more I think about it, the more I am convinced that you should not be allowed to register a subtype that doesnt have an associated format. The design for subtypes having an associated format implies that they there is only a one-to-one relationship between a detector subclass and some format.

Reply:
Agreed.  I changed this, and I did go ahead and make the universal type associate with ``GENERIC``.